### PR TITLE
SPARK-8916 [Documentation, MLlib] Add @since tags to mllib.regression

### PR DESCRIPTION
**Pull request by prayagchandran**
Added @since tags to mllib.regression

### CODE REVIEW

File: mllib/src/main/scala/org/apache/spark/mllib/regression/StreamingLinearAlgorithm.scala [hunk 1]
```diff
@@ -130,7 +147,10 @@ abstract class StreamingLinearAlgorithm[
   }
 
 
-  /** Java-friendly version of `predictOnValues`. */
+  /** Java-friendly version of `predictOnValues`.
```
Comment (JoshRosen) [inline mllib/src/main/scala/org/apache/spark/mllib/regression/StreamingLinearAlgorithm.scala]: This is mis-formatted; the line with the `/**` shouldn't have the method description on it (see the other multi-line Scaladoc strings in this file for examples of the proper style).

Comment (prayagchandran) [inline mllib/src/main/scala/org/apache/spark/mllib/regression/StreamingLinearAlgorithm.scala]: I see. I will correct this to match other Scaladoc strings.

**Comment by prayagchandran** [comment]
@JoshRosen I reformatted the comments now. Also added a few more tags I had missed. Please take a look.

**Comment by JoshRosen** [comment]
Formatting seems fine to me, but I'll let @mengxr or @jkbradley do the final sign-off on this (I'm not really a MLlib reviewer).

Jenkins, this is ok to test.

**Comment by prayagchandran** [comment]
Thanks Josh!

**Comment by mengxr** [comment]
cc @dbtsai

**Comment by mengxr** [comment]
ok to test

**Comment by MechCoder** [comment]
In general: we need not add tags to `equals`, `hashCode`, `toString` and private / protected methods. Please make those changes

File: mllib/src/main/scala/org/apache/spark/mllib/regression/GeneralizedLinearAlgorithm.scala [hunk 2]
```diff
@@ -71,6 +77,8 @@ abstract class GeneralizedLinearModel(val weights: Vector, val intercept: Double
    *
    * @param testData array representing a single data point
    * @return Double prediction from the trained model
+   *
+   * @since 0.8.0
```
Comment (MechCoder) [inline mllib/src/main/scala/org/apache/spark/mllib/regression/GeneralizedLinearAlgorithm.scala]: How were you able to tag this? Any version less that 1.0 gives me fatal: Invalid object name error.

**Comment by jkbradley** [comment]
Update: There is now an annotation which we're using instead of the since tag: [https://issues.apache.org/jira/browse/SPARK-9864]  Would you mind updating this to use the annotation?  Thanks!

**Comment by jkbradley** [comment]
@prayagchandran It will be great to merge this ASAP to include it in the 1.5 RC.  I'll watch for updates.  Thanks!

File: mllib/src/main/scala/org/apache/spark/mllib/regression/GeneralizedLinearAlgorithm.scala (hunk 2 above)
Comment (prayagchandran) [inline mllib/src/main/scala/org/apache/spark/mllib/regression/GeneralizedLinearAlgorithm.scala]: I hadn't checked for the right signature of the method. I will correct this and add other updates to the PR soon.

File: mllib/src/main/scala/org/apache/spark/mllib/regression/IsotonicRegression.scala [hunk 3]
```diff
@@ -148,6 +160,9 @@ class IsotonicRegressionModel (
   /** A convenient method for boundaries called by the Python API. */
   private[mllib] def predictionVector: Vector = Vectors.dense(predictions)
 
+  /*
```
Comment (dbtsai) [inline mllib/src/main/scala/org/apache/spark/mllib/regression/IsotonicRegression.scala]: Should be `/**`

File: mllib/src/main/scala/org/apache/spark/mllib/regression/IsotonicRegression.scala [hunk 4]
```diff
@@ -155,6 +170,9 @@ class IsotonicRegressionModel (
   override protected def formatVersion: String = "1.0"
 }
 
+/*
```
Comment (dbtsai) [inline mllib/src/main/scala/org/apache/spark/mllib/regression/IsotonicRegression.scala]: ditto

File: mllib/src/main/scala/org/apache/spark/mllib/regression/IsotonicRegression.scala [hunk 5]
```diff
@@ -200,6 +218,9 @@ object IsotonicRegressionModel extends Loader[IsotonicRegressionModel] {
     }
   }
 
+  /*
```
Comment (dbtsai) [inline mllib/src/main/scala/org/apache/spark/mllib/regression/IsotonicRegression.scala]: ditto

File: mllib/src/main/scala/org/apache/spark/mllib/regression/Lasso.scala [hunk 6]
```diff
@@ -44,6 +46,9 @@ class LassoModel (
     weightMatrix.toBreeze.dot(dataMatrix.toBreeze) + intercept
   }
 
+  /*
```
Comment (dbtsai) [inline mllib/src/main/scala/org/apache/spark/mllib/regression/Lasso.scala]: ditto

File: mllib/src/main/scala/org/apache/spark/mllib/regression/Lasso.scala [hunk 7]
```diff
@@ -51,8 +56,14 @@ class LassoModel (
   override protected def formatVersion: String = "1.0"
 }
 
+/*
```
Comment (dbtsai) [inline mllib/src/main/scala/org/apache/spark/mllib/regression/Lasso.scala]: ditto

File: mllib/src/main/scala/org/apache/spark/mllib/regression/Lasso.scala [hunk 8]
```diff
@@ -51,8 +56,14 @@ class LassoModel (
...
+ * @since 1.3.0
+ */
 object LassoModel extends Loader[LassoModel] {
 
+  /*
```
Comment (dbtsai) [inline mllib/src/main/scala/org/apache/spark/mllib/regression/Lasso.scala]: ditto

File: mllib/src/main/scala/org/apache/spark/mllib/regression/RegressionModel.scala [hunk 9]
```diff
@@ -24,6 +24,9 @@ import org.apache.spark.api.java.JavaRDD
 import org.apache.spark.mllib.linalg.Vector
 import org.apache.spark.rdd.RDD
 
+/*
```
Comment (dbtsai) [inline mllib/src/main/scala/org/apache/spark/mllib/regression/RegressionModel.scala]: ditto

File: mllib/src/main/scala/org/apache/spark/mllib/regression/RidgeRegression.scala [hunk 10]
```diff
@@ -45,6 +47,9 @@ class RidgeRegressionModel (
     weightMatrix.toBreeze.dot(dataMatrix.toBreeze) + intercept
   }
 
+  /*
```
Comment (dbtsai) [inline mllib/src/main/scala/org/apache/spark/mllib/regression/RidgeRegression.scala]: ditto

File: mllib/src/main/scala/org/apache/spark/mllib/regression/RidgeRegression.scala [hunk 11]
```diff
@@ -52,8 +57,14 @@ class RidgeRegressionModel (
   override protected def formatVersion: String = "1.0"
 }
 
+/*
```
Comment (dbtsai) [inline mllib/src/main/scala/org/apache/spark/mllib/regression/RidgeRegression.scala]: ditto

File: mllib/src/main/scala/org/apache/spark/mllib/regression/RidgeRegression.scala [hunk 12]
```diff
@@ -52,8 +57,14 @@ class RidgeRegressionModel (
...
+ * @since 1.3.0
+ */
 object RidgeRegressionModel extends Loader[RidgeRegressionModel] {
 
+  /*
```
Comment (dbtsai) [inline mllib/src/main/scala/org/apache/spark/mllib/regression/RidgeRegression.scala]: ditto

**Comment by mengxr** [comment]
@dbtsai Do you have time to make a final pass?

**Comment by dbtsai** [comment]
Thanks. Merged into master and 1.5 branch.

**Pull request closed without merging**